### PR TITLE
fix: remove unused promise warning in setNotFoundHandler with preHandler

### PR DESCRIPTION
**Pull request by mcollina**
Fixes https://github.com/fastify/fastify-rate-limit/issues/355

### CODE REVIEW

**Comment by mcollina** [comment]
I don't know how to write a test for this. Our current typing tests have no problem, but the bad eslint warning in https://github.com/fastify/fastify-rate-limit/issues/355 remains.

**Review by gurgunday — APPROVED** [review]

File: types/instance.d.ts [hunk 1]
```diff
@@ -459,7 +459,7 @@ export interface FastifyInstance<
   setNotFoundHandler<RouteGeneric extends RouteGenericInterface = RouteGenericInterface, ContextConfig extends ContextConfigDefault = ContextConfigDefault, TypeProvider extends FastifyTypeProvider = FastifyTypeProviderDefault, SchemaCompiler extends FastifySchema = FastifySchema> (
     opts: {
       preValidation?: preValidationHookHandler<RawServer, RawRequest, RawReply, RouteGeneric, ContextConfig, SchemaCompiler, TypeProvider> | preValidationHookHandler<RawServer, RawRequest, RawReply, RouteGeneric, ContextConfig, SchemaCompiler, TypeProvider>[];
```
Comment (Uzlopak) [inline types/instance.d.ts:461]: seems preValidation also have the same issue.

Comment (mcollina) [reply types/instance.d.ts]: fixed

**Comment by Uzlopak** [comment]
The thing is, that there is afaik no way to implement it properly.

**Review by climba03003 — APPROVED** [review]
As I remember, everytime we apply similar change. It always break something.
But, it still worth a try.

**Pull request merged**